### PR TITLE
[DC-2766]  Separate analytics/cdr_ops/rdr_export_qc.py into two notebooks

### PR DESCRIPTION
**Pull request by jp3477**

### CODE REVIEW

File: data_steward/analytics/cdr_ops/raw_rdr_export_qc.py [hunk 1]
```diff
@@ -0,0 +1,951 @@
...
+#     language: python
+#     name: python3
+# ---
+
+# + tags=["parameters"]
+project_id = ""
+old_rdr = ""
+new_rdr = ""
+raw_rdr = "default"  # do not need to provide this if running on a raw rdr import
+new_rdr_sandbox = ""
```
Comment (brendagutman) [inline data_steward/analytics/cdr_ops/raw_rdr_export_qc.py:21]: I believe this parameter is not needed. There would not be a rdr sandbox at this stage and the queries that use this parameter should be performed after cleaning. I'll mark them below.

File: data_steward/analytics/cdr_ops/raw_rdr_export_qc.py [hunk 2]
```diff
@@ -0,0 +1,951 @@
...
+# According to this [ticket](https://precisionmedicineinitiative.atlassian.net/browse/DC-2488), Question and Answer concepts that are identified in the `old_map_short_codes` table should not be paired with 0-valued concept_identifiers after the RDR dataset is cleaned. These concept identifiers include the `observation_concept_id` and `observation_source_concept_id` fields.
+
+# ## Question Codes
+
+# Check the question codes
+tpl = JINJA_ENV.from_string("""
+WITH question_codes AS (
+  SELECT
+    pmi_code
+  FROM `{{project_id}}.{{sandbox_dataset}}.old_map_short_codes`
```
Comment (brendagutman) [inline data_steward/analytics/cdr_ops/raw_rdr_export_qc.py:728]: Query that might need to be moved to the clean notebook.

File: data_steward/analytics/cdr_ops/raw_rdr_export_qc.py [hunk 3]
```diff
@@ -0,0 +1,951 @@
...
+# -
+
+# ## Answer Codes
+
+# Check the answer codes
+tpl = JINJA_ENV.from_string("""
+WITH answer_codes AS (
+  SELECT
+    pmi_code
+  FROM `{{project_id}}.{{sandbox_dataset}}.old_map_short_codes`
```
Comment (brendagutman) [inline data_steward/analytics/cdr_ops/raw_rdr_export_qc.py:766]: Query that might need to be moved to the clean notebook.

File: data_steward/analytics/cdr_ops/raw_rdr_export_qc.py [hunk 4]
```diff
@@ -0,0 +1,951 @@
...
+#     display_name: Python 3
+#     language: python
+#     name: python3
+# ---
+
+# + tags=["parameters"]
+project_id = ""
+old_rdr = ""
+new_rdr = ""
+raw_rdr = "default"  # do not need to provide this if running on a raw rdr import
```
Comment (brendagutman) [inline data_steward/analytics/cdr_ops/raw_rdr_export_qc.py:20]: I believe this parameter is not needed. There would not be a rdr sandbox at this stage and the queries that use this parameter should be performed after cleaning. I'll mark them below.

File: data_steward/analytics/cdr_ops/raw_rdr_export_qc.py [hunk 5]
```diff
@@ -0,0 +1,951 @@
...
+    """)
+query = tpl.render(new_rdr=new_rdr, project_id=project_id)
+execute(client, query)
+
+# # Identify Questions That Dont Exist in the Cleaned RDR Export
+# This identifies questions as indicated by a PPI vocabulary and Question concept_class_id that
+# do not exist in the cleaned dataset but did exist in the raw dataset.
+
+if raw_rdr == "default":
+    print("Not running code to validate cleaned dataset against raw dataset.")
```
Comment (brendagutman) [inline data_steward/analytics/cdr_ops/raw_rdr_export_qc.py:402]: Most likely not needed in the raw RDR.

File: data_steward/analytics/cdr_ops/raw_rdr_export_qc.py [hunk 6]
```diff
@@ -0,0 +1,951 @@
...
+    )
+        SELECT *
+        from question_codes
+        where concept_id not in (select observation_source_concept_id from used_q_codes)
+        and concept_id in (select observation_source_concept_id from used_rawq_codes)
+        """)
+    query = tpl.render(new_rdr=new_rdr, project_id=project_id, raw_rdr=raw_rdr)
+    execute(client, query)
+
+# # Make sure previously corrected missing data still exists
```
Comment (brendagutman) [inline data_steward/analytics/cdr_ops/raw_rdr_export_qc.py:429]: Most likely not needed in the raw RDR notebook

File: data_steward/analytics/cdr_ops/raw_rdr_export_qc.py [hunk 7]
```diff
@@ -0,0 +1,951 @@
...
+AND {{PIPELINE_TABLES}}.calculate_age(observation_date, EXTRACT(DATE FROM birth_datetime)) < 18
+''')
+query = tpl.render(new_rdr=new_rdr,
+                   project_id=project_id,
+                   PIPELINE_TABLES=PIPELINE_TABLES)
+execute(client, query)
+
+# # Check for missing questionnaire_response_id
+
+# Survey data in the RDR export should all have **questionnaire_response_id**
```
Comment (brendagutman) [inline data_steward/analytics/cdr_ops/raw_rdr_export_qc.py:654]: Most likely not needed in the raw RDR notebook.

File: data_steward/analytics/cdr_ops/raw_rdr_export_qc.py [hunk 8]
```diff
@@ -0,0 +1,951 @@
...
+# -
+
+# ### Question-Answer Codes Combo
+
+# Check that mapped answer codes are paired with correctly mapped question codes.  If the question codes are zero valued, the question and answer pair will be dropped from the clean version of the CDR.
+tpl = JINJA_ENV.from_string("""
+WITH answer_codes AS (
+  SELECT
+    pmi_code
+  FROM `{{project_id}}.{{sandbox_dataset}}.old_map_short_codes`
```
Comment (brendagutman) [inline data_steward/analytics/cdr_ops/raw_rdr_export_qc.py:804]: Most likely not needed in the raw RDR notebook

**Comment by jp3477** [comment]
@brendagutman Thanks for suggesting all of these. I'll implement them all.

**Comment by brendagutman** [comment]
Thanks for the changes @jp3477! The rest looks good to me.

**Review by brendagutman — APPROVED** [review]

**Pull request merged**